libcontainer/setns_init_linux.go

```diff
@@ -131,6 +131,7 @@ func (l *linuxSetnsInit) Init() error {
 
 	// Close the log pipe fd so the parent's ForwardLogs can exit.
 	logrus.Debugf("setns_init: about to exec")
+	_ = l.pipe.Close()
```

Comment:
You are closing this too early. Can do that right before exec. Maybe add a comment why.

Reply:
Or just move it to _before_ "Close the log pipe" comment (similar to how it's done in standard_init_linux.go, except we don't need the debug logging I guess):

https://github.com/opencontainers/runc/blob/02120488a4c0fc487d1ed2867e901eeed7ce8ecf/libcontainer/standard_init_linux.go#L240-L248